source/Calamari.Shared/Deployment/Conventions/ExtractPackage.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.IO;
+using Calamari.Integration.FileSystem;
+using Calamari.Integration.Packages;
+using Calamari.Util;
+
+namespace Calamari.Deployment.Conventions
+{
+    public class ExtractPackage : IExtractPackage
+    {
+        readonly ICombinedPackageExtractor combinedPackageExtractor;
+        readonly ICalamariFileSystem fileSystem;
+        readonly IVariables variables;
+        readonly ILog log;
+
+        public ExtractPackage(ICombinedPackageExtractor combinedPackageExtractor, ICalamariFileSystem fileSystem, IVariables variables, ILog log)
+        {
+            this.combinedPackageExtractor = combinedPackageExtractor;
+            this.fileSystem = fileSystem;
+            this.variables = variables;
+            this.log = log;
+        }
+
+        public void ExtractToStagingDirectory(PathToPackage pathToPackage, IPackageExtractor customPackageExtractor = null)
```

Comment:
This replaces `ExtractPackageToStagingDirectoryConvention` when the `staging` parameter is not supplied